libsawtooth/src/journal/publisher/mod.rs

```diff
@@ -361,7 +361,12 @@ impl BlockPublisher {
 
             let block_id = block.block().header_signature().to_string();
 
-            self.block_broadcaster.broadcast(block)?;
+            let publishing_details = BlockPublishingDetails {
+                injected_batch_ids: candidate_block.injected_batch_ids.into_iter().collect(),
```

Comment:
Can we just use a `HashSet` here too instead of converting it to a `Vec`?

Reply:
If it can be deref'ed to a slice, maybe.

Reply:
Does not look like it can be.  If this struct is part of the public api, I think the method returning a `&[String]` seems more appropriate.